strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -54,6 +69,205 @@ def __init__(self):
         self._supported["mixed_states"] = True
         self._init_modes = None
         self.circuit = None
+        self.ancillae_samples_dict = {}
+
+    def run_prog(self, prog, **kwargs):
+        """Runs a strawberryfields program using the bosonic backend.
+
+        Args:
+            prog (object): sf.Program instance
+
+        Returns:
+            tuple: list of applied commands,
+                    dictionary of measurement samples,
+                    dictionary of ancilla measurement samples
+
+        Raises:
+            NotApplicableError: if an op in the program does not apply to the bosonic backend
+            NotImplementedError: if an op in the program is not implemented in the bosonic backend
+        """
+        from strawberryfields.ops import (
+            Bosonic,
+            Catstate,
+            DensityMatrix,
+            Fock,
+            GKP,
+            Ket,
+            MbSgate,
+            _New_modes,
+        )
+
+        # Initialize the circuit. This applies all non-Gaussian state-prep
+        self.init_circuit(prog)
+
+        # Apply operations to circuit. For now, copied from LocalEngine;
+        # only change is to ignore preparation classes and ancilla-assisted gates
+        # TODO: Deal with Preparation classes in the middle of a circuit.
+        applied = []
+        samples_dict = {}
+        all_samples = {}
+        for cmd in prog.circuit:
+            nongausspreps = (Bosonic, Catstate, DensityMatrix, Fock, GKP, Ket, _New_modes)
+            ancilla_gates = (MbSgate,)
+            # For ancilla-assisted gates, if they return measurement values, store
+            # them in ancillae_samples_dict
+            if type(cmd.op) in ancilla_gates:
```

Comment:
Better to make this a call to `isinstance`, which is a lot safer that `type`:

```suggestion
            if isinstance(cmd.op, ancilla_gates):
```